### PR TITLE
test: add test for nexmark q6 group topn

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/frontend/planner_test/tests/testdata/output/nexmark_watermark.yaml [hunk 1]
```diff
@@ -513,6 +513,133 @@
...
+    Table 4294967294 { columns: [ seller, avg, $expr2 ], primary key: [ $2 ASC, $0 ASC ], value indices: [ 0, 1, 2 ], distribution key: [ 0 ], read pk prefix len hint: 2 }
+
+  eowc_stream_plan: |-
+    StreamMaterialize { columns: [seller, avg, $expr2(hidden)], stream_key: [$expr2, seller], pk_columns: [$expr2, seller], pk_conflict: NoCheck }
+    └─StreamProject { exprs: [$expr4, (sum / count::Decimal) as $expr7, $expr2] }
+      └─StreamEowcOverWindow { window_functions: [sum($expr6) OVER(PARTITION BY $expr4 ORDER BY $expr1 ASC ROWS BETWEEN 10 PRECEDING AND CURRENT ROW), count($expr6) OVER(PARTITION BY $expr4 ORDER BY $expr1 ASC ROWS BETWEEN 10 PRECEDING AND CURRENT ROW)] }
+        └─StreamSort { sort_column_index: 2 }
+          └─StreamExchange { dist: HashShard($expr4) }
+            └─StreamProject { exprs: [$expr4, $expr6, $expr1, $expr2], output_watermarks: [$expr1] }
+              └─StreamGroupTopN [append_only] { order: [$expr6 ASC], limit: 1, offset: 0, group_key: [0, 1], output_watermarks: [$expr1] }
```
Comment (st1page) [inline src/frontend/planner_test/tests/testdata/output/nexmark_watermark.yaml:620]: I think the groupTopN's watermark derivation is wrong :thinking:

**Review by st1page — APPROVED** [review]
LGTM

File: src/frontend/planner_test/tests/testdata/input/nexmark_source.yaml [hunk 2]
```diff
@@ -160,6 +160,25 @@
     ) AS Q;
   expected_outputs:
   - planner_error
+- id: nexmark_q7_group_top1
```
Comment (lmatz) [inline src/frontend/planner_test/tests/testdata/input/nexmark_source.yaml:163]: `q6` ?

File: src/frontend/planner_test/tests/testdata/output/nexmark.yaml [hunk 3]
```diff
@@ -497,6 +497,81 @@
 
     Table 4294967294 { columns: [ auction, num, window_start, window_start#1 ], primary key: [ $0 ASC, $2 ASC, $3 ASC ], value indices: [ 0, 1, 2, 3 ], distribution key: [ 2 ], read pk prefix len hint: 3 }
 
+- id: nexmark_q7_group_top1
```
Comment (lmatz) [inline src/frontend/planner_test/tests/testdata/output/nexmark.yaml:500]: q6?

File: src/frontend/planner_test/tests/testdata/input/nexmark_source.yaml (hunk 2 above)
Comment (st1page) [reply src/frontend/planner_test/tests/testdata/input/nexmark_source.yaml:163]: :hot_face:

**Pull request merged**